Redesign sentences

trang opened the pull request:
This pull request is part of #1180. It also addresses #1324.

The sentences have been redesigned for non-authenticated users.

* The icons to translate, add to favorite, add to list have been removed. For non-authenticated users, they just redirect to a page to log in. Although they do give a hint to users what are the features of Tatoeba, I think they mostly pollute the UI which is why they were removed.
* There are now a explicit labels for the sentence (`Sentence {number} - belongs to {user}`), the translations (`Translations`) and the indirect translations (`Translations of translations`). Hopefully users will understand better why some translations are in grey.
* Users no longer navigate to the sentence by clicking on the `#` or arrow icons, but on the info icon that is at the end of each sentence/translation.
* An arrow icon has been added in front of each translation in order to help distinguish between direct and indirect translations when there are a lot of translations. Direct translations have a black arrow. Indirect translations have a grey arrow.
* The text of the indirect translations are kept in black for better readability. There is now a label that says explicitly "Translations of translations", so there is less need to have it grey.
* Unapproved sentences are displayed the same as before (grey + warning icon), with a tooltip that says `This sentence is not reliable.` when hovering over the sentence. I'm not very satisfied with this tooltip text, but I'd like to find something else than "unapproved".
* The language icon is kept on the left side. We cannot have it on the right side because the language icon is a pillar in the structure of the sentences. It helps users find more easily the translations in the lanugages they are more specifically interested. And for sentences that have several translations in a same language, it helps users identify more quickly which blocks of translations are in which language.
* The sentence in the comments will be re-adapted later on to be consistent with the new sentence design. It's not in the scope of this pull request.

Authenticated users still have by default the old design, but can activate the new one from their settings if they wish to try to it out. Some pages will still keep the old design no matter what the setting is:

* The random sentence on the homepage for authenticated users, because the feature to load another random sentence doesn't work yet with the new design. Non-authenticated users have the new design of the random sentence, but do not have the reload option.
* The sentence details page, because the new sentence design does not have the sentence menu (translate, edit, etc), and users should still have access to a page where they can access every sentence feature without switching back and forth between old and new design. Also, the audio icon was not re-added (yet?) in the new design, so the sentence details would allow to listen to the audio of a specific sentence.

The new design is notably faster to render than the old one. For instance someone who has their settings set to 100 sentences per page will have some noticeable lag while searching sentences (the page will stay grey for a few seconds, or even freeze). The new design doesn't have this lag. So for users who mostly browse Tatoeba and use the sentence menu only punctually, it can be more advantageous to switch to the new design.

The features of the sentence menu will be introduced into the new design progressively. This will probably take until end of December. Once every feature has been implemented in the new design, the old design will be removed for everyone.

ckjpn commented:
Perhaps also changing the indirect translation’s text to gray might be a good idea -- something light enough that it's obviously not the same as direct translations, but still dark enough to be easy to read.
![screen shot 2016-10-14 at 07 09 16](https://cloud.githubusercontent.com/assets/7934273/19368978/c20b0102-91dd-11e6-8f16-f3590b5fe645.png)

(perhaps between #888 ~ #aaa)

ckjpn commented:
> Once every feature has been implemented in the new design, the old design will be removed for everyone.

Perhaps, at least temporarily, at that time you could make the old way an option, just in case some members would still prefer the old way, and/or just in case the new way doesn't work exactly like expected.

I often add sentences to lists from the "browse" page, and I also find it very useful to see which sentences I have rated OK and can also add ratings while browsing.  Usually I only jump to the sentence's detailed page if I need to add a tag.

If you could also make it possible to add tags from the "browse" page it would save me a lot of time.  However, that might add too much clutter for most people.  Maybe you could make it an option in the settings.

Perhaps in the settings, you could make it an option to only use the "cut-down version" that you are currently using.

Maybe, you could also only use this "cut-down / light" version for non-logged in members, since they are only browsing and don't need the full set of "tools".  This might help the pages load faster for those who don't use the tools.  If a high percentage of page views were using a "light" version, perhaps it would also make the website faster for everybody else, too.

trang commented:
> Perhaps also changing the indirect translation’s text to gray might be a good idea

I'll keep it with the same color as the translations for now. I'll first have to see if the label `Translations of translations` and the color of the arrow are enough or not.

> Perhaps, at least temporarily, at that time you could make the old way an option, just in case some members would still prefer the old way, and/or just in case the new way doesn't work exactly like expected.

There will most likely be one or two weeks where users will still have by default the old UI even if the new UI is completely ready. But once we've switched, there will be no option to go back to the old sentence. If something doesn't work as expected by then, it will be fixed in the new UI.

> Maybe, you could also only use this "cut-down / light" version for non-logged in members.

As I've said in the description, the sentences have been redesigned for **non-authenticated users** (= non-logged in). It was only towards the end that I implemented the option for logged in users to optionally use the new design, if they wish to. But the new design is aimed at first for non-logged in users, and is part of the optimization mentioned in #1324.
